aiida/calculations/transfer.py

```diff
@@ -0,0 +1,211 @@
+    RemoteData nodes must be from the same machine).
+
+    The user needs to provide three inputs:
+    * `instructions`: a dict node specifying which files to copy from which nodes.
+    * `source_nodes`: a dict of nodes, each with a unique identifier label as its key.
+    * `metadata.computer`: the computer that contains the remote files and will contain the final
+    RemoteData node.
+
+    The `instructions` dict must have the `retrieve_files` flag. The CalcJob will create a new
+    folder in the remote machine (RemoteData) and put all the files there and will either:
+        (1) leave them there (`retrieve_files = False`) or ...
+        (2) create a local FolderData and copy there all the files (`retrieve_files = True`)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tupples of 3 that have the following format:
+
+        [ ..., ( source_node_label, path_to_file_in_source, path_to_file_in_target), ... ]
+
+    The two main lists to include are `local_files` (for files to be taken from FolderData nodes)
+    and `remote_files` (for files to be taken from RemoteData nodes). Alternatively, files inside
+    of RemoteData nodes can instead be put in the `symlink_files` list: the only difference is that
+    files from the first list will be fully copied in the target RemoteData folder, whereas for the
+    files in second list only a symlink to the original file will be created there. This will only
+    affect the content of the final RemoteData target folder, but in both cases the full file will
+    be copied back in the local target FolderData (if `retrieve_files = True).
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input(
+            'instructions',
+            valid_type=orm.Dict,
+            help='A dictionary containing the `retrieve_files` flag and at least one of the file lists:'
+            '`local_files`, `remote_files` and/or `symlink_files`.',
+        )
+        spec.input_namespace(
+            'source_nodes',
+            valid_type=(orm.FolderData, orm.RemoteData),
+            dynamic=True,
+            help='All the nodes that contain files referenced in the instructions.',
+        )
+
+        # The transfer just needs a computer, the code are resources are set here
+        spec.inputs['code'].default = None
+        spec.inputs['code'].valid_type = (orm.Code, type(None))
+        spec.inputs['metadata']['computer'].required = True
+        spec.inputs['metadata']['options']['resources'].default = {
+            'num_machines': 1,
+            'num_mpiprocs_per_machine': 1,
+        }
+
+        # Retrieved node should only be generated if actually retrieving files
+        #spec.outputs[cls.link_label_retrieved].required = False
+        #spec.output(cls.link_label_retrieved).required = False
```

Comment:
Noup; I didn't change this because I seem to remember we briefly mentioned this in our last meeting but went on with other issues and didn't decide what to finally do with this. Or if we did I forgot / didn't write it down, sorry =S.